server/src/main/java/org/tctalent/server/api/portal/CandidateExamPortalApi.java

```diff
@@ -0,0 +1,69 @@
+package org.tctalent.server.api.portal;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.tctalent.server.api.admin.ITableApi;
+import org.tctalent.server.exception.EntityExistsException;
+import org.tctalent.server.exception.InvalidRequestException;
+import org.tctalent.server.exception.NoSuchObjectException;
+import org.tctalent.server.model.db.CandidateExam;
+import org.tctalent.server.request.candidate.exam.CreateCandidateExamRequest;
+import org.tctalent.server.request.candidate.exam.UpdateCandidateExamsRequest;
+import org.tctalent.server.service.db.CandidateExamService;
+import org.tctalent.server.util.dto.DtoBuilder;
+
+@RestController
+@RequestMapping("/api/portal/candidate-exam")
+public class CandidateExamPortalApi implements ITableApi<CreateCandidateExamRequest,UpdateCandidateExamsRequest,UpdateCandidateExamsRequest> {
+
+  private final CandidateExamService candidateExamService;
+
+  @Autowired
+  public CandidateExamPortalApi(CandidateExamService candidateExamService) {
+    this.candidateExamService = candidateExamService;
+  }
+
+  @PostMapping("/{candidateId}")
+  public @NotNull Map<String, Object> createCandidateExam(@Valid @RequestBody CreateCandidateExamRequest request, @PathVariable Long candidateId)
+      throws InvalidRequestException, NoSuchObjectException {
```

Comment:
You need to override the relevant interface methods - that is what standardizes the API.

For example, this method should implement the IJoinedTableApi.create method. That defines the standard method name "create" (not "createCandidateExam"), the type of mapping (Post) and defines any path mappings - parentId in this case .

Reply:
Hi Ehsan

You need to change the declaration of CandidateExamPortalApi to...

public class CandidateExamPortalApi implements **_IJoinedTableApi_**<...

Note also that the first parameter of ITableApi and IJoinedTableApi is the SEARCH request, not the CreateRequest

Then you need to implement the methods defined in the interface.

I will record a little video showing what I mean.

Reply:
Thanks @camerojo ,
Pushed new changes.